### ISSUE TITLE
Support location.hash in BrowserHistory

### ISSUE THREAD

**Issue by jackmoore** [rackt/react-router#1167]
For example, if I call transitionTo with the following:
```
this.transitionTo('http://localhost:3333/blog/4?test=true#frag')
```

I am transitioned to this:
```
http://localhost:3333/blog/4?test=true%23frag
```

Is this a bug, or is there some reason why I shouldn't be able to pass a URL like that to transitionTo?

Got the same result testing in React Router 0.12.2 and 0.13.3.

**Comment by mjackson** [comment]
The `transitionTo` API on current master has the signature `transitionTo(pathname, query, state)` where both `query` and `state` are optional. So, you probably need to do a `transitionTo('/blog/4', { test: 'true#frag' })`.

@jackmoore Can you please try this on current master? I'm pretty sure it will work, especially since I just merged #1265.

**Comment by taion** [comment]
`transitionTo` doesn't exist any more and `history` supports hashes now.
